.gitlab-ci.yml

```diff
@@ -113,9 +110,29 @@ test:
   needs:
     - check-std
   script:
-    - for crate in ${ALL_CRATES}; do
-        cargo test --verbose --all-features --release --manifest-path ${crate}/Cargo.toml;
-      done
+    - cargo test --verbose --all-features --no-fail-fast --release --workspace
+
+codecov:
+  stage:                           workspace
+  <<:                              *docker-env
+  needs:
+    - check-std
+  variables:
+    # according to https://github.com/mozilla/grcov/blob/master/README.md we have to use these variables, however
+    CARGO_INCREMENTAL:             0
+    # it does not work with -Zno-landing-pads
+    RUSTFLAGS:                     "-Zprofile -Ccodegen-units=1 -Cinline-threshold=0 -Clink-dead-code -Coverflow-checks=off"
+    # it's better to use `cargo-taurpalin` instead of `grcov`, but the former doesn't work within docker properly
```

Comment:
please mention that using `tarpaulin` requires a docker configuration to be less secure (or whatever is actually the reason).